src/layer.h

```diff
@@ -144,15 +144,17 @@ class Layer
 
 // layer factory function
 typedef Layer* (*layer_creator_func)();
+typedef void (*layer_destroyer_func)(Layer*);
 
 struct layer_registry_entry
 {
 #if NCNN_STRING
     // layer type name
-    const char* name;
+    std::string name;
```

Comment:
what is the rationale of using non-POD type here

Reply:
in c/c++, string compiled to .text section, and we can use it(all the same string have same address) in program anywhere.
but in python, if a string object have no reference(no name), it's reference count is 0 after pass to c/c++, and recycled by gc, for example:
`net.register_custom_layer("YoloV5Focus", YoloV5Focus_layer_creator)`
during this call, net can see the string "YoloV5Focus", but after this call, "YoloV5Focus" is recycled, and net.load_param can't find this registed layer by name.

so anyother solutions?